File: TwoPointers/MaxArea.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxi = -1;
        int start = 0;
        int end = height.size() - 1;
        while(start < end)
        {
            if(height[start] > height[end])
            {
                maxi = max(maxi, height[end]*(end-start));
                end--;
            }
            else
            {
                maxi = max(maxi, height[start]*(end-start));
                start++;
            }   
        }
        return maxi;
    }
};

File: TwoPointers/MaxArea.py
class Solution(object):
    def maxArea(self, height):
        maxi = -1
        start = 0
        end = len(height) - 1
        for i in range(len(height)):
            if height[start] > height[end]:
                maxi = max(maxi,height[end]*(end-start))
                end -= 1
            else:
                maxi = max(maxi,height[start]*(end-start))
                start += 1
        return maxi